lit-base/model_service.py

```diff
@@ -0,0 +1,91 @@
+import mxnet as mx
+import tempfile
+import urllib2
+import numpy as np
+from PIL import Image
+
+from model_loader import sym, arg_params, aux_params
+
+from collections import namedtuple
+from StringIO import StringIO
+
+
+ctx = mx.cpu()
+
+
+def preprocess(event):
+    """
+    Preprocess Lambda event into input data format for mx mod.
+    :return: input data NDArray
+    event : dict
+        Lambda event
+    """
+    img = None
+    try:
+        # GET method
+        if event.get('httpMethod') == 'GET':
+            url = event['queryStringParameters']['url']
+            # download image from url
+            img_file = tempfile.NamedTemporaryFile(delete=True)
+            if url:
+                req = urllib2.urlopen(url)
```

Comment:
I think we need to wrap a retry logic here to make it more robust

Reply:
I slightly prefer fail the request directly under Lambda context for two reasons:
1. Lambda has limited time and resources to execute, and should succeed or fail fast when it's triggered
2. Retry could easily expose customer to extra charge than they expect, and make them vulnerable to malicious attack

Reply:
Adding a retry block does not violate the Lambda resource limitation. It will make this service more robust. E.g. in the case of overloading, you apparently just don't want to just break downstream service right away.

Reply:
Thanks! Modified accordingly. Since concern 

2. Retry could easily expose customer to extra charge than they expect, and make them vulnerable to malicious attack

is still valid, I only sleep it for a short time and retry once